packages/client/README.md

```diff
@@ -109,10 +109,24 @@ If the provider you're looking for hasn't been created yet, see the [develop a p
 
 Once you've added a provider as a dependency, it can be registered with OpenFeature like this:
 
+#### Awaitable
+
+To register a provider and ensure it is ready before further actions are taken, you can use the `setProviderAndWait` method as shown below:
+
+```ts
+await OpenFeature.setProviderAndWait(new MyProvider());
+```  
+
+#### Synchronous
```

Comment:
Should it be Asynchronous here ?

Reply:
I really struggled with the name here. Techically, the method is synchronous but the provider may not be ready immediately. Calling it asynchronous would imply that it would return a promise or accept a callback. However, I do agree that this is confusing. Can you think of a better term we could use?

Reply:
Yes, the naming is super complex on this one.
We can use something like **blocking** / **non blocking** maybe ?

Reply:
I personally think this is accurate. The method returns immediately. The fact is starts some asynchronous jobs doesn't make the method itself asynchronous.